### DISCUSSION TITLE
How is the table order determined (while inspecting the database)?

### DISCUSSION THREAD

**Discussion by frankvollebregt** [simolus3/drift#1671] [Q&A] [answered]
When I use things like the `drift_db_viewer` on the device or the `storage_inspector` for remote debugging, they show the tables in a specific order. This order is not alphabetic, nor are the tables I defined in the same files grouped together in this list.

Is there a rationale behind this ordering? It makes finding the right tables in the list a bit of a struggle while using these plugins/packages to inspect my database.

**Comment by simolus3** [comment, marked as answer]
The generator will sort tables topologically, meaning that a table will appear before tables that reference it (e.g. through a foreign key constraint). This is useful since iterating over tables in migrations will automatically create them in the right order. There's no guarantee on the order of tables outside of it matching references, I would call this an implementation detail :D

For tools like `storage_inspector` or `drift_db_viewer`, it might make sense to sort tables alphabetically instead of this internal order. cc @vanlooverenkoen and @NicolaVerbeeck, have you considered this as an option?

**Reply by NicolaVerbeeck** [reply]
@simolus3 alphabetical sorting has been implemented as of today in the local storage inspector version 😊

**Reply by frankvollebregt** [reply]
Thanks @simolus3, it makes sense to have this as the ordering internally.

And thank you @NicolaVerbeeck for such a fast turnaround time on the feature request :)

**Reply by vanlooverenkoen** [reply]
@simolus3  & @frankvollebregt drift_db_viewer is also updated. In fact every app that uses `db_viewer` will have alphabetic table sorting
